pkg/netlink/netlink_watcher.go

```diff
@@ -0,0 +1,1998 @@
+}
+
+// rtnScope map of string key as RTN scope
+var rtnScope = map[string]int{
+	"global":  unix.RT_SCOPE_UNIVERSE,
+	"site":    unix.RT_SCOPE_SITE,
+	"link":    unix.RT_SCOPE_LINK,
+	"local":   unix.RT_SCOPE_HOST,
+	"nowhere": unix.RT_SCOPE_NOWHERE,
+}
+
+var testFlag = map[int]string{
+	unix.RTNH_F_ONLINK:    "onlink",
+	unix.RTNH_F_PERVASIVE: "pervasive",
+}
+
+//--------------------------------------------------------------------------
+//###  Neighbor Database Entries
+//--------------------------------------------------------------------------
+
+// linkTable wg sync.WaitGroup
+var linkTable []vn.Link
+
+// vrfList netlink libarary var
+var vrfList []vn.Link
+
+// deviceList netlink libarary var
+var deviceList []vn.Link
+
+// vlanList netlink libarary var
+var vlanList []vn.Link
+
+// bridgeList netlink libarary var
+var bridgeList []vn.Link
+
+// vxlanList netlink libarary var
+var vxlanList []vn.Link
+
+// linkList netlink libarary var
+var linkList []vn.Link
+
+// NameIndex netlink libarary var
+var NameIndex = make(map[int]string)
+
+// oldgenmap old map
+var oldgenmap = make(map[interface{}]interface{})
+
+// latestgenmap latest map
+var latestgenmap = make(map[interface{}]interface{})
+
+// notifyEvents array
+var notifyEvents = []string{"_added", "_updated", "_deleted"}
+
+//--------------------------------------------------------------------------
+//###  NexthopStruct Database Entries
+//--------------------------------------------------------------------------
+
+// l2NhIDCache
+var l2NhIDCache = make(map[l2NexthopKey]int)
+
+// nhIDCache Variable
+var nhIDCache = make(map[nexthopKey]int)
+
+const (
+	// Define each route type as a constant
+	routeTypeConnected = "connected"
+	routeTypeEvpnVxlan = "evpn-vxlan"
+	routeTypeStatic    = "static"
+	routeTypeBgp       = "bgp"
+	routeTypeLocal     = "local"
+	routeTypeNeighbor  = "neighbor"
+)
+
+const (
+	// RouteAdded event const
+	RouteAdded = "route_added"
+	// RouteUpdated event const
+	RouteUpdated = "route_updated"
+	// RouteDeleted event const
+	RouteDeleted = "route_deleted"
+	// NexthopAdded event const
+	NexthopAdded = "nexthop_added"
+	// NexthopUpdated event const
+	NexthopUpdated = "nexthop_updated"
+	// NexthopDeleted event const
+	NexthopDeleted = "nexthop_deleted"
+	// FdbEntryAdded event const
+	FdbEntryAdded = "fdb_entry_added"
+	// FdbEntryUpdated event const
+	FdbEntryUpdated = "fdb_entry_updated"
+	// FdbEntryDeleted event const
+	FdbEntryDeleted = "fdb_entry_deleted"
+	// L2NexthopAdded event const
+	L2NexthopAdded = "l2_nexthop_added"
+	// L2NexthopUpdated event const
+	L2NexthopUpdated = "l2_nexthop_updated"
+	// L2NexthopDeleted event const
+	L2NexthopDeleted = "l2_nexthop_deleted"
+)
+
+// getFlag gets the flag
+func getFlag(s string) int {
+	f := 0
+	for ff, ss := range testFlag {
+		if s == ss {
+			f |= ff
+		}
+	}
+	return f
+}
+
+// getFlagString return flag of type string
+
+func getFlagString(flag int) string {
+	for flg, str := range testFlag {
```

Comment:
testFlag[flag]

Reply:
flg is integer which is being checked against the passed int value.

Reply:
`testFlag` is `map[int]string` We don't have to traverse all elements. We can use lookup
```Go
str, ok := testFlag[flag]
if !ok {
  return ""
}
return str
```

Reply:
done